components/brave_wallet/renderer/js_cardano_provider.cc

```diff
@@ -134,20 +128,17 @@ void JSCardanoProvider::OnEnableResponse(
 
   v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
   if (!error) {
-    gin::Handle<JSCardanoWalletApi> wallet_api = gin::CreateHandle(
-        isolate, new JSCardanoWalletApi(
-                     std::move(remote), base::PassKey<JSCardanoProvider>(),
-                     global_context.Get(isolate), isolate, render_frame()));
-    if (wallet_api.IsEmpty()) {
-      return;
-    }
-    v8::Local<v8::Value> wallet_api_value = wallet_api.ToV8();
-    v8::Local<v8::Object> wallet_api_object;
-    if (!wallet_api_value->ToObject(context).ToLocal(&wallet_api_object)) {
-      return;
-    }
+    JSCardanoWalletApi* wallet_api =
+        cppgc::MakeGarbageCollected<JSCardanoWalletApi>(
+            isolate->GetCppHeap()->GetAllocationHandle(), std::move(remote),
+            base::PassKey<JSCardanoProvider>(), global_context.Get(isolate),
+            isolate, render_frame());
+
+    v8::Local<v8::Object> wallet_api_object =
+        wallet_api->GetWrapper(isolate).ToLocalChecked();
```

Comment:
Is this call to ToLocalChecked() guaranteed not to CHECK()?

Reply:
I think what Oleg has to confirm here is the life time for this, because we shouldn't put an `if` around here "just in case". We need to know for sure if we need an `if` and if we do, why.

Reply:
@cypt4 Do you mind creating a follow-up issue to investigate these?